src/Uno.UI/UI/Xaml/Documents/LanguageTagCache.skia.cs

```diff
@@ -0,0 +1,28 @@
+﻿using System;
+using System.Collections.Generic;
+using Windows.Globalization;
+
+#nullable enable
+
+namespace Windows.UI.Xaml.Documents;
+
+internal class LanguageTagCache
+{
+	[ThreadStatic] private static readonly Dictionary<string, string[]> _languageTagsCache = new();
+
+	public static string[] GetLanguageTag(TextElement element)
+	{
+		var key = element.Language
+			?? ApplicationLanguages.PrimaryLanguageOverride
+			?? string.Empty;
+		if (!_languageTagsCache.TryGetValue(key, out var tag))
+		{
+			var cultureInfo = key.Length == 0
+				? global::System.Globalization.CultureInfo.CurrentUICulture
+				: global::System.Globalization.CultureInfo.GetCultureInfoByIetfLanguageTag(key);
```

Comment:
I think this can throw an exception.

Reply:
I do not agree. When PrimaryLanguageOverride is empty string the first language present in ManifestLanguages is used.
This is a workaround to avoid unnnecessary call to ApplicationLanguages.ManifestLanguages.First().

https://github.com/unoplatform/uno/blob/baa80040e6862968524f34959e631406ca1482fc/src/Uno.UWP/Globalization/ApplicationLanguages.cs#L62-L105